### DISCUSSION TITLE
Caching schema to AST and restoring resolvers

### DISCUSSION THREAD

**Discussion by andris-sevcenko** [webonyx/graphql-php#1095] [Q&A] [answered]
I'm trying to improve the time we spend on building the Schema - I have the usual things down - lazy loading, type loader, and all that jazz. I'm now attempting to incorporate saving the Schema as AST, loading it, and re-attaching the resolvers. I'm trying to do this following [the issue here](https://github.com/webonyx/graphql-php/issues/310) and the [docs here](https://webonyx.github.io/graphql-php/schema-definition-language/#defining-resolvers).

However.

If there'a Query object defined like this:

```php
$query = new ObjectType([
  'name' => 'Query',
  'fields' => [
    'foo' => [
      'type' => Type::listOf(FooInterface::getType()),
      'args' => FooArguments::getArguments(),
      'resolve' => FooResolver::class . '::resolve',
      'complexity' => FooComplexityFunction () {},
    ],
  ])
]);
```

I'm having trouble reattaching this resolver. This is because `entries` is a field of type `[FooInterface]` and the type decorator function does not get invoked for `foo` because it's not a type. Also, a different field might be of type `FooInterface`, so a different resolver would apply (one that returns a single item). This means I can't really attach a resolver to `FooInterface` and just go with that.

Currently, it seems like the only way to do this would be to construct a `resolveField` function instead and use tha. Does that sound like the right approach here? 

If so, (and this is somewhat related, because it's annoying when people ask two things in one issue) - does this also imply that complexity values cannot be restored this way?

**Comment by spawnia** [comment, marked as answer]
If you are defining your schema in code, there is no need to convert it to the AST representation of the SDL. That will not save time.

**Reply by andris-sevcenko** [reply]
Gotcha. Re-reading the documentation page made it click for me. Thank you!
